src/plugins/unified_field_list/public/components/field_stats/field_top_values_bucket.tsx

```diff
@@ -21,31 +21,51 @@ import type { IFieldSubTypeMulti } from '@kbn/es-query';
 import type { DataViewField } from '@kbn/data-views-plugin/common';
 import type { AddFieldFilterHandler } from '../../types';
 
-export interface FieldTopValuesBucketProps {
-  type?: 'normal' | 'other';
+export interface FieldTopValuesBucketParams {
   field: DataViewField;
   fieldValue: unknown;
   formattedFieldValue?: string;
   formattedPercentage: string;
   progressValue: number;
   count: number;
   color: string;
+  type?: 'normal' | 'other';
+}
+
+export type OverrideFieldTopValueBarCallback = (
+  params: FieldTopValuesBucketParams
+) => Partial<FieldTopValuesBucketParams>;
+
+export interface FieldTopValuesBucketProps extends FieldTopValuesBucketParams {
   'data-test-subj': string;
   onAddFilter?: AddFieldFilterHandler;
+  /**
+   * Optional callback to allow overriding props on bucket level
+   */
+  overrideFieldTopValueBar?: OverrideFieldTopValueBarCallback;
 }
 
-export const FieldTopValuesBucket: React.FC<FieldTopValuesBucketProps> = ({
-  type = 'normal',
-  field,
-  fieldValue,
-  formattedFieldValue,
-  formattedPercentage,
-  progressValue,
-  count,
-  color,
+const FieldTopValuesBucket: React.FC<FieldTopValuesBucketProps> = ({
```

Comment:
Already discussed with @qn895 , but adding for reference, testing against the `apm-7.13.2-transaction-eden-hist-000001`, the percentages are adding up to over 100% in many cases. This is happening inside APM, and on `main` in Discover too:

<img width="670" alt="image" src="https://user-images.githubusercontent.com/7405507/196983463-39196672-a6ce-44df-8a43-0dc4809fecfb.png">

<img width="439" alt="image" src="https://user-images.githubusercontent.com/7405507/196983767-096be6d7-52a3-401d-a540-d5b3f2f62390.png">

Reply:
Are those fields arrays? Then occurrences of values can start overlapping among records and it won't add up to 100%.

Reply:
Although I am not sure how one item can go over 100%. Could you please compare results with Field Statistics table on Discover page?

Reply:
Looks like some of them are arrays, although all the docs I'm inspecting there is only ever one item in the array. Here's an example (on `main`):

<img width="1295" alt="image" src="https://user-images.githubusercontent.com/7405507/199766446-7d94d01f-1a9d-419b-8d8b-8e0dff6afbba.png">

Reply:
> Looks like some of them are arrays, although all the docs I'm inspecting there is only ever one item in the array. Here's an example (on `main`):
> 

Any field in ES can have one or more values regardless of the type. I think that's why Discover displays every field as an array. 
Wrt `transaction.result` I'm 99% sure that it'll only ever have a single value. But I'll double-check with the APM Server team.

Reply:
@sqren @peteharverson I think this is [an issue with the random sampler itself](https://github.com/elastic/elasticsearch/issues/77575) as it's reproducible on main via API in Discover, Lens, etc. For these fields specifically, they are not just normal values, not arrays. Will escalate the issue up on the Elasticsearch side as it's impacting more plugins.

Reply:
This issue is now fixed via https://github.com/elastic/kibana/pull/144962

Reply:
With the fix https://github.com/elastic/kibana/pull/144962, the % values in the popover now look good for me - posting same example as above:

<img width="861" alt="image" src="https://user-images.githubusercontent.com/7405507/201725709-39fc7fee-92c8-415e-8b32-e507ec07528c.png">